src/black/parsing.py

```diff
@@ -84,14 +85,18 @@ def lib2to3_parse(src_txt: str, target_versions: Iterable[TargetVersion] = ()) -
             result = drv.parse_string(src_txt, True)
             break
 
-        except ParseError as pe:
-            lineno, column = pe.context[1]
+        except (TokenError, ParseError) as err:
+            if isinstance(err, ParseError):
+                lineno, column = err.context[1]
+            else:
+                lineno, column = err.args[1]
```

Comment:
I'm trying to understand why i'm getting linenumber value as 2 for token error:
```
(black) bash-3.2$ python -m src.black --check --code "print("
error: cannot format <string>: Cannot parse: 2:0: <line number missing in source>
(black) bash-3.2$ python -m src.black --check --code "print([)" 
error: cannot format <string>: Cannot parse: 1:7: print([)
```

Reply:
Yeahhh that seems to be a bug with the blib2to3 library. We can just ignore that for this PR, it's better if we can just get this landed so while the error message won't be perfect, it should be better than `error: cannot format <string>: ('EOF in multi-line statement', (2, 0))` :)

Although a problem is that the source isn't printed at all which is unfortunate since it would be quite useful. Perhaps it would be good to also include that it was an "unexpected EOF" error in the message?

Good catch tho!